clang/tools/driver/cc1_main.cpp

```diff
@@ -413,11 +503,34 @@ Optional<int> CompileJobCache::initialize(CompilerInstance &Clang) {
     MCOutputID = ID;
     return Error::success();
   };
-  bool ComputedJobNeedsReplay = CacheOpts.WriteOutputAsCASID || UseCASBackend;
 
-  CacheBackend = std::make_unique<ObjectStoreCachingOutputs>(
-      Clang, CacheOpts.WriteOutputAsCASID, UseCASBackend,
-      ComputedJobNeedsReplay, MCOutputID, CAS, Cache);
+  if (!CacheOpts.CompilationCachingServicePath.empty()) {
+    Expected<remote_cache::ClientServices> Clients =
+        remote_cache::createCompilationCachingRemoteClient(
+            CacheOpts.CompilationCachingServicePath);
+    if (!Clients)
+      return reportCachingBackendError(Clang.getDiagnostics(),
+                                       Clients.takeError());
+    if (CacheOpts.WriteOutputAsCASID)
+      return reportCachingBackendError(
+          Clang.getDiagnostics(),
```

Comment:
This diagnosis and the one below should be emitted in Frontend when validating options.

Reply:
Good idea! See https://github.com/apple/llvm-project/pull/5355/commits/a5b724b9b53c2d2c84d5708237ce137f1d40f5ca